### PR TITLE
Post notification when user is fetched / created / updated

### PR DESCRIPTION
**Pull request by yuliaveres**
Instead of retrieving with completion handler

### CODE REVIEW

File: Token/Controllers/Settings/SettingsController.swift [hunk 1]
```diff
@@ -106,6 +108,10 @@ open class SettingsController: UITableViewController {
 
         self.fetchAndUpdateBalance()
     }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: .currentUserUpdated, object: nil)
```
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: This is automatic now in Swift. 💪🏽

File: Token/Models/TokenUser.swift [hunk 2]
```diff
@@ -141,13 +146,13 @@ public class TokenUser: NSObject, NSCoding {
             let deserialised = (try? JSONSerialization.jsonObject(with: userData, options: [])),
             var json = deserialised as? [String: Any] {
 
-            // Because of payment address migration, we have to override the stored payment address.
+            // Because of payment address pmigration, we have to override the stored payment address.
```
Comment (Elland) [inline Token/Models/TokenUser.swift]: Typo? 👍

File: Token/Controllers/Settings/SettingsController.swift (hunk 1 above)
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: yeah

Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: i got used to this a lot :D removing

File: Token/Controllers/Settings/SettingsController.swift [hunk 3]
```diff
@@ -97,6 +97,8 @@ open class SettingsController: UITableViewController {
 
     open override func viewDidLoad() {
         super.viewDidLoad()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(updateUI), name: .currentUserUpdated, object: nil)
```
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: `currentUserDidUpdate` would be more idiomatic but this is fine. 👍

File: Token/Controllers/Settings/SettingsController.swift [hunk 4]
```diff
@@ -115,6 +117,13 @@ open class SettingsController: UITableViewController {
         self.usernameLabel.text = TokenUser.current?.displayUsername
         self.userAvatarImageVIew.image = TokenUser.current?.avatar
     }
+    
+    @objc private func updateUI() {
```
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: Function name could be clearer, but this works.

**Comment by Elland** [comment]
Looks fine here. 💥✨

**Pull request merged**